src/github3/actions/secrets.py

```diff
@@ -0,0 +1,227 @@
+    but not its actual value.
+
+    .. attribute:: name
+
+        The name of the secret
+
+    .. attribute:: created_at
+
+        The timestamp of when the secret was created
+
+    .. attribute:: updated_at
+
+        The timestamp of when the secret was last updated
+    """
+
+    class_name = "RepositorySecret"
+
+
+class SharedOrganizationSecret(_Secret):
+    """An object representing an organization secret for GitHub Actions that is
+    shared with the repository.
+
+    See https://docs.github.com/en/rest/actions/secrets for more details.
+    GitHub never reveals the secret value through its API, it is only accessible
+    from within actions. Therefore, this object represents the secret's metadata
+    but not its actual value.
+
+    .. attribute:: name
+
+        The name of the secret
+
+    .. attribute:: created_at
+
+        The timestamp of when the secret was created
+
+    .. attribute:: updated_at
+
+        The timestamp of when the secret was last updated
+    """
+
+    class_name = "SharedOrganizationSecret"
+
+
+class OrganizationSecret(_Secret):
+    """An object representing am organization secret for GitHub Actions.
+
+    See https://docs.github.com/en/rest/actions/secrets for more details.
+    GitHub never reveals the secret value through its API, it is only accessible
+    from within actions. Therefore, this object represents the secret's metadata
+    but not its actual value.
+
+    .. attribute:: name
+
+        The name of the secret
+
+    .. attribute:: created_at
+
+        The timestamp of when the secret was created
+
+    .. attribute:: updated_at
+
+        The timestamp of when the secret was last updated
+    """
+
+    class_name = "OrganizationSecret"
+
+    def _update_attributes(self, secret):
+        super()._update_attributes(secret)
+        self.visibility = secret["visibility"]
+        if self.visibility == "selected":
+            self._selected_repos_url = secret["selected_repositories_url"]
+
+    def selected_repositories(self, number=-1, etag=""):
+        """Iterates over all repositories this secret is visible to.
+
+        :param int number:
+            (optional), number of repositories to return.
+            Default: -1 returns all selected repositories.
+        :param str etag:
+            (optional), ETag from a previous request to the same endpoint
+        :returns:
+            Generator of selected repositories or None if the visibility of this
+            secret is not set to 'selected'.
+        :rtype:
+            :class:`~github3.repos.ShortRepository`
+        """
+        from .. import repos
+
+        if self.visibility != "selected":
+            return None
+
+        return self._iter(
+            int(number),
+            self._selected_repos_url,
+            repos.ShortRepository,
+            etag=etag,
+            list_key="repositories",
+        )
+
+    def set_selected_repositories(self, repository_ids: typing.List[int]):
```

Comment:
Prefer `Sequence` over `List` as a `List[int]` precludes `Tuple[int]` or any other sequence like data that Python will happily correctly serialize to a JSON Array